src/jdk.incubator.vector/share/classes/jdk/incubator/vector/X-Vector.java.template

```diff
@@ -605,8 +605,10 @@ public abstract class $abstractvectortype$ extends AbstractVector<$Boxtype$> {
                 return blend(broadcast(-1), compare(NE, 0, m));
             }
 #if[BITWISE]
-            if (op == NOT || op == NEG) {
-                return blend(lanewise(op), m);
+            if (op == NOT) {
+                return lanewise(XOR, broadcast(-1), m);
+            } else if (op == NEG) {
+                return lanewise(NOT, m).lanewise(ADD, broadcast(1), m);
```

Comment:
Why not stick to the same pattern is the non-masked `lanewiseTemplate` method?
```suggestion
            if (op == NOT) {
                return broadcast(-1).lanewise(XOR, this, m);
            } else if (op == NEG) {
                return broadcast(0).lanewise(SUB, this, m);
```

Reply:
Seems the result is not equal to `this` vector for the non-active lanes (false lane for `m`)?

Reply:
Yes, result will not be same since broadcasted zero vector is first operand. We need to select the lanes of 'this' vector when corresponding mask bit is false.

Reply:
Doh! Can you place a comment the NOT/ADD expression? e.g.
```
// Cannot perform broadcast(0).lanewise(SUB, this, m)
// since we need to retain unselected lanes in this vector  
```